frontend/online-shop/src/App.tsx

```diff
@@ -0,0 +1,31 @@
+import { CssBaseline } from "@material-ui/core";
+import { ThemeProvider } from "@material-ui/styles";
+import {
+  BrowserRouter as Router,
+  Switch,
+  Route,
+  Redirect,
+} from "react-router-dom";
+import "./App.css";
+import { ProductDetail } from "./components/ProductDetail";
+import ProductList from "./components/ProductList";
+import MyTheme from "./theme/MyTheme";
+
+export const App = () => {
+  return (
+    <ThemeProvider theme={MyTheme}>
+      <CssBaseline>
+        <Router>
+          <Switch>
+            <Route path="/product/:number" component={ProductDetail}></Route>
```

Comment:
You should be more specific with the path parameters. Either /product/:id or /product/:productId.

Reply:
Fixed.